### ISSUE TITLE
MS Edge does not set map container as focusable

### ISSUE THREAD

**Issue by jplante** [Leaflet/Leaflet#3772]
It appears as though keyboard navigation with the Edge browser is not working.
From my initial testing, using Windows 10:

navigating to http://leafletjs.com/:

**Chrome 44.0.2403.157 m**
Keyboard navigation works as expected, panning the map in all directions.

**Edge**
Keyboard navigation does not work as expected, instead up/down moves the page scroll-bar

After testing with LeafletJS 1.0-beta2, I see the same behavior.

We have seen this issue on multiple machines, with and without touch screens.
Has anyone else been able to verify that this is actually the case?

**Comment by IvanSanchez** [comment]
By any chance do you have caret mode active?

If you run `L.DomEvent.on(document, 'keydown', function(ev){console.log(ev.keyCode);});` in Edge, and then pres arrow up/down, do you get keycodes 38 and 40, or something entirely different?

**Comment by jplante** [comment]
I get key codes 39, 38, 40 etc.
Learning about what caret mode is now, but I, and the other testers of this have not done any unique modifications to our Edge browser settings.

**Comment by IvanSanchez** [comment]
I'm going to make a wild guess: IE is losing map focus of the map because of #3314, and as a result the keyboard event handler gets detached.

**Comment by IvanSanchez** [comment]
OK, I've got something. Edge returns `0` when asking for the `tabIndex` value of a non-focusable HTMLElement with no explicit `tabindex` HTML attribute, whereas the rest of the web browsers returns -1.

This means https://github.com/Leaflet/Leaflet/blob/master/src/map/handler/Map.Keyboard.js#L33 doesn't make the container focusable (because Edge reports it is already focusable), so the keyboard event listeners are never attached.

@mourner Changing line 33 into `tabIndex === -1 || tabIndex === 0` should work but feels... dirty. Any other ideas?

**Comment by IvanSanchez** [comment]
BTW, upstream bug report at https://connect.microsoft.com/IE/feedbackdetail/view/1711605/tabindex-attribute-getter-returns-0-instead-of-1-for-elements-without-explicit-tabindex

**Comment by mourner** [comment]
@IvanSanchez actually I don't remember why there's a check there. Should we just remove it?

**Comment by IvanSanchez** [comment]
@mourner No, I guess some developers might actually use a greater-than-zero tabindex on their map containers for accesibility purposes.

**Comment by mourner** [comment]
Then something like `<= 0` should be fine.
